src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,141 @@
+package duke.task;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Represents a task. A <code>Task</code> object contains a description,
+ * a  boolean representing whether or not the task has been done and additional information.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    protected Type type;
+    protected String info;
+    protected Date date;
+
+    /**
+     * Empty constructor for Task
+     */
+    public Task() {
+
+    }
+
+
+    /**
+     * Constructs a new Task.
+     * @param description description of the task
+     * @param isDone whether or not the task has been done
+     */
+    public Task(String description, boolean isDone, String info) {
+        this.description = description.trim();
+        this.isDone = isDone;
+        this.info = info.trim();
+    }
+
+    /**
+     * Returns a boolean representing whether or not the task is done.
+     * @return true if the task is done, false otherwise
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Returns a string which is the first character of each task type
+     * @return a "T", "D", or "E"
+     */
+    public String getSymbol() {
+        switch (type){
+        case TODO:
+            return "T";
+        case DEADLINE:
+            return "D";
+        case EVENT:
+            return "E";
+        default:
+            return "";
+        }
+    }
+
+    /**
+     * Returns a boolean representing whether or not the task is done.
+     * @return true if the task is done, false otherwise
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Overrides toString method
+     * @return a String in the to-be-displayed format
+     */
+    @Override
+    public String toString() {
+        if (info.equals("")) {
+            return "[" + getSymbol() + "][" + getStatusIcon() + "] " + description;
+        } else {
+            String[] infos = info.split(" ", 2);
+            return "[" + getSymbol() + "][" + getStatusIcon() + "] " + description + " (" + infos[0] +":  " +  infos[1] + ")";
+        }
+    }
+
+    /**
+     * Returns a String in the format for file saving. An abstract method to be implemented in child classes.
+     */
+    public abstract String getFileStringFormat();
+
+    /**
+     * Checks if the string is in date format dd/MM/yyyy HHmm and returns the parsed form of the  date.
+     * @param str the task's info String
+     * @return true if the task is done, false otherwise
+     */
+    public String checkDate(String str)  {
```

Comment:
Handling of dates is very organised! I manually coded the String output of every day and month but using existing Java libraries makes code much neater! Good job!